File: 27-remove-element/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        unordered_map<int,int> freq;
        int n = nums.size();
        for(int i=0;i<n;i++){
            if(nums[i] != val){
                freq[nums[i]]++;
                // cout << nums[i] << " ";
            }
        }

        int i=0;
        for(auto& it : freq){
            cout << it.first << "->" <<it.second<<" ";
            while(it.second--){
                nums[i] = it.first;
                i++;
            }
        }
        
        return i;
    }
};

File: 27-remove-element/remove-element.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        count = 0
        n = len(nums)
        i = 0
        while i < len(nums):
            if nums[i] == val:
                count += 1
                nums.pop(i)
            else:
                i += 1
        return n - count
